docs/development.txt

```diff
@@ -0,0 +1,82 @@
+[[development-process]]
+= JanusGraph Development Process
+
+The following sections describe the JanusGraph development process.
+
+
+[[development-decisions]]
+== Development Decisions
+
+Many development decisions will be made during the day-to-day work of JanusGraph contributors without any extra process overhead.
+However, for larger bodies of work and significant updates and additions, the following process shall be followed.
+Significant work may include, but is not limited to:
+
+* A major new feature or subproject, e.g., a new storage adapter
+* Incrementing the version of a core dependency such as a Apache TinkerPop
+* Addition or deprecation of a public API
+* Internal shared data structure or API change
+* Addition of a new major dependency
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the GitHub issue tracker
+* Start a DISCUSS thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called
+* Two +1 votes are required for the change to be accepted
+
+[[branching]]
+== Branching
+
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their own branches, and submitting pull requests when ready. If multiple developers wish to collaborate on a feature,
+they may request that a feature branch be created in JanusGraph repository.  If the developers are not committers, a JanusGraph committer will be assigned as the shepherd for that branch.  The shepherd will be responsible for merging pull requests to that branch.
+
+=== Branch Naming Conventions
+
+All branch names shall be prepended with `Issue_#_`.
```

Comment:
For example, if there was an issue 114 to add Tupl support to JanusGraph, and people needed to collaborate on this branch, then it could be named `Issue_114_Tupl`. This naming convention only applies to hosted branches created for collaboration, not to branches in forks created for normal pull requests.